tests/Kernel/EuLoginEventSubscriberTest.php

```diff
@@ -112,61 +131,63 @@ public function testValidationParameters(): void {
     $session = $this->container->get('session');
     $session->start();
 
-    // Add the ticket to the options that will be passed to the URL request
-    // and add the session to the request we just made.
-    $options['query'] = $query;
-    $uri = '/casservice';
-    /** @var \Symfony\Component\HttpFoundation\Request $request */
-    $request = Request::create(Url::fromUri('base:' . $uri, $options)->toString());
-    $request->setSession($session);
+    // Returns the query string parameter from the last executed request.
+    $get_query_from_request = function () use ($session, $query): array {
+      $request = Request::create(Url::fromUri('base:/casservice', ['query' => $query])->toString());
+      $request->setSession($session);
+      $this->container->get('http_kernel')->handle($request);
+      $last_request = end($this->history)['request'];
```

Comment:
We could still put a `@var` doc on `$last_request`, so that static analysis understands the `->getUri()` below.
Up to you.

Reply:
Done